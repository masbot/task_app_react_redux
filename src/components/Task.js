import React, { Component } from 'react';

class Task extends Component {
    render() {
        return (
            <div className="task-item">
                <div>{this.props.task}</div>
                {/* <i className="fa fa-pencil edit" aria-hidden="true" /> */}
                <i className="fa fa-trash-o delete btn-remove" onClick={() => this.props.remove(this.props.id)} aria-hidden="true" />
            </div>
        )
    }
}

export default Task;