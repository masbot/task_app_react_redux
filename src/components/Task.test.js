import React from 'react';
import { shallow } from 'enzyme';
import Task from './task';

describe('Task', () => {
    const mockRemoveTask = jest.fn();
    const props = { remove: mockRemoveTask };
    const task = shallow(<Task {...props}/>);

    it('renders correctly', () => {
        expect(task).toMatchSnapshot();
    });

    describe('when user wants to remove task', () => {
        beforeEach(() => {
            task.find('.btn-remove').simulate('click');
        });

        it('calls the removeTask callback', () => {
            expect(mockRemoveTask).toHaveBeenCalled();
        });
    })

});