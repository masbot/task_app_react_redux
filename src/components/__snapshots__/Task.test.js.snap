// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Task
    remove={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div />,
        <i
          aria-hidden="true"
          className="fa fa-trash-o delete btn-remove"
          onClick={[Function]}
/>,
      ],
      "className": "task-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "className": "fa fa-trash-o delete btn-remove",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div />,
          <i
            aria-hidden="true"
            className="fa fa-trash-o delete btn-remove"
            onClick={[Function]}
/>,
        ],
        "className": "task-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "className": "fa fa-trash-o delete btn-remove",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
